homeassistant/components/backup/config.py

```diff
@@ -214,6 +301,26 @@ async def _create_backup(now: datetime) -> None:
                 password=config_data.create_backup.password,
             )
 
+            # Delete old backups more numerous than copies
+            if config_data.delete_after_config.copies is None:
+                return
+
+            def _backups_filter(backups: dict[str, Backup]) -> dict[str, Backup]:
+                """Return oldest backups more numerous than copies to delete."""
+                # we need an extra check here since we await before
+                # this filter is applied
+                if config_data.delete_after_config.copies is None:
+                    return {}
+                return dict(
+                    sorted(
+                        backups.items(),
+                        key=lambda backup_item: backup_item[1].date,
+                        reverse=True,
```

Comment:
Not sure why we reverse here. We should be able to sort ascending and keep the first (oldest) part when slicing.